packages/clarity-js/src/interaction/baseline.ts

```diff
@@ -0,0 +1,49 @@
+import { Event, BooleanFlag } from "@clarity-types/data";
+import { BaselineData, BaselineState } from "@clarity-types/interaction";
+import { time } from "@src/core/time";
+import encode from "./encode";
+
+export let state: BaselineState = null;
+let buffer: BaselineData = null;
+let update: boolean = false;
+
+export function start(): void {
+    reset();
+}
+
+export function reset(): void {
+    state = update ? { time: time(), event: Event.Baseline, data: buffer } : state;
+    buffer = buffer ? buffer : { visible: BooleanFlag.True, docWidth: 0, docHeight: 0, scrollX: 0, scrollY: 0, pointerX: 0, pointerY: 0 };
+    update = false;
+}
+
+export function visibility(visible: string): void {
+    buffer.visible = visible === "visible" ? BooleanFlag.True : BooleanFlag.False;
+    update = true;
+}
+
+export function track(event: Event, x: number, y: number): void {
+    switch (event) {
```

Comment:
can you explain how this helps in  sending previous state?
i.e - from my understanding - we are tracking changes to pointer and sending the last state as a part of the current upload, i don't see how we are resending it in the next payload?